src/Microsoft.ML.Data/Transforms/ConvertTransform.cs

```diff
@@ -433,246 +352,240 @@ private static bool TryCreateEx(IExceptionContext ectx, ColInfo info, DataKind k
 
-            public TypeInfoCommand(IHostEnvironment env, Arguments args)
-            {
-                Contracts.CheckValue(env, nameof(env));
-                _host = env.Register(LoadName);
-                _host.CheckValue(args, nameof(args));
-            }
+                // Ensure that the conversion is legal. We don't actually cache the delegate here. It will get
+                // re-fetched by the utils code when needed.
+                if (!Conversions.Instance.TryGetStandardConversion(srcType.ItemType, itemType, out Delegate del, out bool identity))
+                    return false;
 
-            private struct TypeNaInfo
-            {
-                public readonly bool HasNa;
-                public readonly bool DefaultIsNa;
+                typeDst = itemType;
+                if (srcType.IsVector)
+                    typeDst = new VectorType(itemType, srcType.AsVector);
 
-                public TypeNaInfo(bool hasNa, bool defaultIsNa)
-                {
-                    HasNa = hasNa;
-                    DefaultIsNa = defaultIsNa;
-                }
+                return true;
             }
 
-            private sealed class KindSetComparer : IEqualityComparer<ISet<DataKind>>
+            public override Schema.Column[] GetOutputColumns()
             {
-                public bool Equals(ISet<DataKind> x, ISet<DataKind> y)
-                {
-                    Contracts.AssertValueOrNull(x);
-                    Contracts.AssertValueOrNull(y);
-                    if (x == null || y == null)
-                        return (x == null) && (y == null);
-                    return x.SetEquals(y);
-                }
-
-                public int GetHashCode(ISet<DataKind> obj)
+                var result = new Schema.Column[_parent._columns.Length];
+                for (int i = 0; i < _parent._columns.Length; i++)
                 {
-                    Contracts.AssertValueOrNull(obj);
-                    int hash = 0;
-                    if (obj != null)
+                    var builder = new Schema.Metadata.Builder();
+                    var srcType = InputSchema[_srcCols[i]].Type;
+                    if (_types[i].IsKnownSizeVector)
+                        builder.Add(InputSchema[ColMapNewToOld[i]].Metadata, name => name == MetadataUtils.Kinds.SlotNames);
+                    if (srcType.ItemType.IsKey && _types[i].ItemType.IsKey &&
+                        srcType.ItemType.KeyCount > 0 &&
+                        srcType.ItemType.KeyCount == _types[i].ItemType.KeyCount)
+                        builder.Add(InputSchema[ColMapNewToOld[i]].Metadata, name => name == MetadataUtils.Kinds.KeyValues);
+                    if (srcType.ItemType.IsNumber && _types[i].ItemType.IsNumber)
+                        builder.Add(InputSchema[ColMapNewToOld[i]].Metadata, name => name == MetadataUtils.Kinds.IsNormalized);
+                    if (srcType.IsBool && _types[i].ItemType.IsNumber)
                     {
```

Comment:
>&& [](start = 75, length = 2)

these guys go on the next line I believe. Also use curlies for the one-liner below, if the condition is longer that one line